sparta/sparta/resources/Pipeline.hpp

```diff
@@ -49,15 +57,15 @@ namespace sparta
      * -# Perform manual or automatic pipeline update (i.e. forward progression). The registered
      *    pipeline stage handlers are called under-the-hood whenever valid pipeline data arrives.
      */
-    template <typename DataT>
+    template <typename DataT, typename EventT=PhasedUniqueEvent>
```

Comment:
You might need to include the DataT in the EventT event:
```
typename EventT=PhasedUniqueEvent<DataT>
```

Reply:
But PhasedUniqueEvent doesn't have template parameter?

Reply:
Right!  Duh.